### PR TITLE
Receive additional Product Properties from GRS

### PR DESCRIPTION
**Pull request by dm-ps**
refs: https://github.com/metasfresh/metasfresh/issues/12300

### CODE REVIEW

File: backend/de.metas.cucumber/src/test/java/de/metas/cucumber/stepdefs/externalreference/S_ExternalReference_StepDef.java [hunk 1]
```diff
@@ -99,6 +111,65 @@ public void metasfresh_contains_s_external_reference(@NonNull final DataTable da
...
+
+		final List<JsonMetasfreshId> bPartnerIds = queryBL.createQueryBuilder(I_C_BPartner.class)
+				.addOnlyActiveRecordsFilter()
+				.create()
+				.setOrderBy(orderBy)
+				.list()
+				.stream()
+				.map(bPartner -> JsonMetasfreshId.of(bPartner.getC_BPartner_ID())).collect(Collectors.toList());
+
+		dataTableEntries.forEach(dataTableEntry -> {
```
Comment (metas-ts) [inline backend/de.metas.cucumber/src/test/java/de/metas/cucumber/stepdefs/externalreference/S_ExternalReference_StepDef.java:142]: No offense, but I wonder what you folks see in `.forEach`...i mean, how is is better than a normal for-loop at this point?

File: backend/de.metas.adempiere.adempiere/migration/src/main/sql/postgresql/system/10-de.metas.adempiere/5620800_sys_gh12300_CBPartnerProduct_ExcludedFromPurchase.sql [hunk 2]
```diff
@@ -0,0 +1,426 @@
...
+INSERT INTO AD_Element_Trl (AD_Language,AD_Element_ID, CommitWarning,Description,Help,Name,PO_Description,PO_Help,PO_Name,PO_PrintName,PrintName,WEBUI_NameBrowse,WEBUI_NameNew,WEBUI_NameNewBreadcrumb, IsTranslated,AD_Client_ID,AD_Org_ID,Created,Createdby,Updated,UpdatedBy,IsActive) SELECT l.AD_Language, t.AD_Element_ID, t.CommitWarning,t.Description,t.Help,t.Name,t.PO_Description,t.PO_Help,t.PO_Name,t.PO_PrintName,t.PrintName,t.WEBUI_NameBrowse,t.WEBUI_NameNew,t.WEBUI_NameNewBreadcrumb, 'N',t.AD_Client_ID,t.AD_Org_ID,t.Created,t.Createdby,t.Updated,t.UpdatedBy,'Y' FROM AD_Language l, AD_Element t WHERE l.IsActive='Y'AND (l.IsSystemLanguage='Y' OR l.IsBaseLanguage='Y') AND t.AD_Element_ID=580468 AND NOT EXISTS (SELECT 1 FROM AD_Element_Trl tt WHERE tt.AD_Language=l.AD_Language AND tt.AD_Element_ID=t.AD_Element_ID)
+;
+
+-- 2022-01-06T18:08:31.939Z
+-- I forgot to set the DICTIONARY_ID_COMMENTS System Configurator
+UPDATE AD_Element_Trl SET Name='Sperre Kaufen Grund', PrintName='Sperre Kaufen Grund',Updated=TO_TIMESTAMP('2022-01-06 20:08:31','YYYY-MM-DD HH24:MI:SS'),UpdatedBy=100 WHERE AD_Element_ID=580468 AND AD_Language='de_CH'
```
Comment (metas-ts) [inline backend/de.metas.adempiere.adempiere/migration/src/main/sql/postgresql/system/10-de.metas.adempiere/5620800_sys_gh12300_CBPartnerProduct_ExcludedFromPurchase.sql:233]: please replace 'Sperre Kaufen Grund' with 'Sperre Einkauf Grund'

**Review by metas-ts — CHANGES_REQUESTED** [review]

File: backend/de.metas.cucumber/src/test/java/de/metas/cucumber/stepdefs/externalreference/S_ExternalReference_StepDef.java (hunk 1 above)
Comment (dm-ps) [reply backend/de.metas.cucumber/src/test/java/de/metas/cucumber/stepdefs/externalreference/S_ExternalReference_StepDef.java]: Changed to normal for-each loop

File: backend/de.metas.adempiere.adempiere/migration/src/main/sql/postgresql/system/10-de.metas.adempiere/5620800_sys_gh12300_CBPartnerProduct_ExcludedFromPurchase.sql (hunk 2 above)
Comment (dm-ps) [reply backend/de.metas.adempiere.adempiere/migration/src/main/sql/postgresql/system/10-de.metas.adempiere/5620800_sys_gh12300_CBPartnerProduct_ExcludedFromPurchase.sql:233]: done

**Review by metas-ts — APPROVED** [review]

**Pull request merged**